FirebaseStorage/Sources/FIRStorageReference.m

```diff
@@ -215,6 +215,13 @@ - (FIRStorageUploadTask *)putFile:(NSURL *)fileURL
 - (FIRStorageUploadTask *)putFile:(NSURL *)fileURL
                          metadata:(nullable FIRStorageMetadata *)metadata
                        completion:(nullable FIRStorageVoidMetadataError)completion {
+  if ([fileURL frs_hasDirectoryPath]) {
```

Comment:
Should we check that it is a file versus not a directory?  Like https://github.com/firebase/firebase-ios-sdk/pull/5742/files?

Reply:
I will play with this. Maybe instead of the check looking like this:
```
 if fileUrl is a directory -> return error
```

We could instead have something like?:

```
 if !(fileUrl is a file) -> return error
```

Reply:
Good point, I think the 2nd option is preferable because there may be other types of URLs that are not handled now, e.g. a symlink or not a file system URL.

Reply:
SGTM